ssl/ssl_test.cc

```diff
@@ -10071,6 +10525,150 @@ TEST(SSLTest, NameLists) {
   }
 }
 
+// Test a successful round-trip for KemKeyShare
```

Comment:
Are these tests operating at the right abstraction level? All the other tests seem to be using `CreateClientAndServer/ConnectClientAndServer/CompleteHandshakes`. I see this is why you changed the visibility in internal.h. I think these are good tests to have, but are they testing something different than what actually happens?

Reply:
If it is valuable to test the failure paths of how we handle unexpected input of the new HybridKeyShare and KemKeyShare classes that we are adding, then I'd argue that these tests are at the correct abstraction level. We still have coverage of the high level "happy path" using the API's you mentioned in [PerformHybridHandshakeTest](https://github.com/aws/aws-lc/pull/1201/files#diff-39cb80e1b51b8f538bf9667decca8d5f25849138ec5623ddece489113dc3c601R11767), but we also have coverage that these SSLKeyShares will fail on unexpected input how we expect them to. 

If the AWS-LC team would prefer an alternate approach that can test these function without exporting them, or not to test these error paths at all, I can also make those changes.